core/src/main/java/org/infinispan/CacheStream.java

```diff
@@ -244,8 +254,24 @@
     * described in more detail in the {@link CacheStream} documentation</p>
     * <p>This intermediate iterator operation will be performed locally and remotely requiring possibly a subset of
     * all elements to be in memory</p>
+    * <p>Any subsequent intermediate operations and the terminal operation are then performed locally.</p>
     * @return the new stream
     */
    @Override
    Stream<R> distinct();
+
+   /**
+    * {@inheritDoc}
+    * <p>Note when using a distributed backing cache for this stream the collector must be marshallable.  This
+    * prevents the usage of {@link java.util.stream.Collectors} class.  However you can use the
+    * {@link org.infinispan.stream.CacheCollectors} static factory methods to create a collector that is created
```

Comment:
Yeah unfortunately the collector provided from CacheCollectors contains a serializable supplier that creates a non serializable collector.  Any suggestion for the wording? :)